server/punishments.ts

```diff
@@ -1158,11 +1158,10 @@ export const Punishments = new class {
 
 		return IPTools.lookup(ip).then(({dnsbl, host, hostType}) => {
 			user = connection.user || user;
-			if (user.locked === '#hostfilter') user.locked = null;
 
-			if (hostType === 'proxy' && !user.trusted && !user.locked) {
+			if (hostType === 'proxy' && !user.locked) {
```

Comment:
Why was `!user.trusted` removed?

Reply:
The only place I see ``Punishments.checkIp`` called is in ``socketConnect`` and the user can't be trusted at this point since it is before ``handleRename``.

Reply:
I put the sanity checks back in in case this function is used differently in the future.